### PR TITLE
Ember cli visualeditor

### PR DESCRIPTION
**Pull request by oliver----**
Replaced VisualEditor integration with ember-cli addon (https://github.com/substance/ember-cli-visualeditor.git).

- reduced configuration in Tahi for VE assets
- introduced a custom, hackable toolbar (see paper/edit/template.hbs)
- VE is integrated on a much lower, modular level as before (e.g., model view separation)
- did not port manuscript placeholders as they will be tackled together with the work on manuscript structure

### CODE REVIEW

**Comment by rizwanreza** [comment]
Nice to see this. :+1: 

I am getting this error: 
bug

**Comment by oliver----** [comment]
I will just do a clean clone and see if it works.
The relevant location is https://github.com/substance/ember-cli-visualeditor/blob/master/addon/initializers/initialize_visual_editor.js#L79-L80 which should load from different URLs as in the error.

**Comment by oliver----** [comment]
ATM I am struggling with phantomjs.
Beyond a problem with the assets route ('/assets/tahi' in app vs '/assets' in test), phantomjs seems to have some problems with the bundled js files I created in ember-cli-visualeditor.
Tests run without probs in browser.

Does anybody have a hint?

**Comment by oliver----** [comment]
This turned out to be somewhat nasty: when running tests ember served empty files for the lazy-loaded scripts - likely some race conditions as when I curled the URL it was served properly. Dunno. Now, I follow the approach to ember-cli package (app.import) the assets when in test environment.

**Comment by oliver----** [comment]
Test suite is passing in my local setup.

**Comment by rizwanreza** [comment]
Rebuilding this on Semaphore.

**Comment by oliver----** [comment]
@rizwanreza does it work for you locally?

**Comment by rizwanreza** [comment]
Not really, seeing this: 
screen shot 2015-02-09 at 10 29 04 am

**Comment by oliver----** [comment]
@rizwanreza Hmmm, bad... maybe this doesn't work https://github.com/substance/tahi/blob/ember-cli-visualeditor/client/Brocfile.js#L54-L57 the same on OSX (?)

Could you check how the output of 'ember build' in /client/ looks liks?

**Comment by oliver----** [comment]
... i.e. there should be folder 'ember-cli-visualeditor' under assets...

**Comment by rizwanreza** [comment]
```
$ ember build
version: 0.1.12
Could not find watchman, falling back to NodeWatcher for file system events
Built project successfully. Stored in "dist/".
```

**Comment by rizwanreza** [comment]
I can see `client/dist/assets/ember-cli-visualeditor/`

**Comment by oliver----** [comment]
Thanks. Can u run tahi and try to reach some of the assets there manually e.g., /assets/ember-cli-visualeditor/visualEditor.css

**Comment by oliver----** [comment]
Do you have time for a that, and maybe we should continue chat on slack?

**Comment by oliver----** [comment]
@tonywok @rizwanreza I found this in the ruby spec helpers: https://github.com/Tahi-project/tahi/blob/master/spec/support/pages/edit_paper_page.rb#L66-L87

Can you explain me to what extend you need to interact with the editing API from the specs and what technical means are possible, i.e., only executeScript and DOM manipulation?

ATM, I use a mocked version of the ve-API for testing as I needed to get around ES5 troubles in PhantomJS. So if you want we could define a specific contract how things look like created by this mock. 

Plz help :)

**Comment by oliver----** [comment]
Rebased onto current master.

**Comment by rizwanreza** [comment]
Good stuff Oliver! 

Here's what I am seeing right now:  
screen shot 2015-02-11 at 4 12 17 pm

Is that okay? There is some issue with font awesome it seems.

**Comment by oliver----** [comment]
Not at all ok.

Let me check...

**Comment by oliver----** [comment]
This is how it looks for me:

image

Hmmm.. any suggestions how to narrow this down?

**Comment by oliver----** [comment]
Could you check if vendor.css does look any similar:

image

**Comment by rizwanreza** [comment]
It looks the same here. 

We are trying to debug this as well. It seems that fonts are not being compiled in the `../fonts` directory. If I change the files to say `'font-name'` without `'../fonts'`, it works fine.

**Comment by oliver----** [comment]
That is strange.

**Comment by rizwanreza** [comment]
We can look into it tomorrow. Go take some :zzz:.

**Comment by oliver----** [comment]
Almost there ;)

**Comment by rizwanreza** [comment]
haha!

**Comment by neo-tahi** [comment]
[ ] Review this PR

File: client/Brocfile.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
 /* global require, module */
 
 var EmberApp = require('ember-cli/lib/broccoli/ember-app');
+var mergeTrees = require('broccoli-merge-trees');
+var pickFiles = require('broccoli-static-compiler');
 
-var app = new EmberApp({storeConfigInMeta: false});
```
Comment (tonywok) [inline client/Brocfile.js]: What's happening here? The `storeConfigInMeta` option is required for `ember-cli-rails`

**Comment by tonywok** [comment]
Saw you needed to increase the build timeout to 30 seconds. Is that still necessary?

File: client/Brocfile.js (hunk 1 above)
Comment (oliver----) [inline client/Brocfile.js]: I wanted to pull all configuration together in one place... i.e., using config/environment.js. Moved the storeConfigInMeta option there (https://github.com/Tahi-project/tahi/pull/816/files#diff-3cd1a229b31460d78e7e154670278c55R20)

**Comment by oliver----** [comment]
Im my case yes. I have the feeling that ember-cli performs much worse than plain broccoli.
It started to be that slow since exporting the ve/i18n assets :(

**Comment by tonywok** [comment]
Yeah, it could be a number of factors.

How big is the actual visual editor library, excluding the ember-cli shim that makes it "tahi aware"?

Perhaps we could include the shim as part of the package, and pull in an already minified version of the visual editor library either via `getScript`. It might also be worth exploring what the impact of including the assets only if building for production - falling back to an ajax request in development?

**Comment by rizwanreza** [comment]
@oliver---- I think it's probably fine to include a basic Editor for now, without all the other localizations for example.

**Comment by oliver----** [comment]
@tonywok @rizwanreza ok... I will try to reduce the assets to a minimum for dev.

**Pull request merged**